src/Surfnet/ServiceProviderDashboard/Application/Service/AttributeService.php

```diff
@@ -90,31 +91,75 @@ public function getUrns(): array
         return $urns;
     }
 
-    public function createEntityDetailAttributes(AttributeList $manageAttributes): array
-    {
+    public function createEntityDetailAttributes(
+        AttributeList $manageAttributes,
+        string $entityType
+    ): array {
         $attributes = [];
         foreach ($manageAttributes->getAttributes() as $attribute) {
             $attributeDto = $this->attributeRepository->findOneByName($attribute[0]->getName());
             if ($attributeDto instanceof AttributeDto) {
                 $viewObject = new EntityDetailAttribute();
                 $viewObject->value = $attribute[0]->getMotivation();
-                $viewObject->informationPopup = $attributeDto->detail->languages[$this->language]->info;
-                $viewObject->label = $attributeDto->detail->languages[$this->language]->label;
-                $viewObject->excludedFor = $attributeDto->form->excludeFrom;
+                $viewObject->informationPopup = $this->getInfoFromAttributeDto($attributeDto, $entityType);
+                $viewObject->label = $this->getLabelFromAttributeDto($attributeDto, $entityType);
+                $viewObject->excludedFor = $attributeDto->excludeOnEntityType;
                 $attributes[] = $viewObject;
             }
         }
         return $attributes;
     }
 
-    public function getAttributeFriendlyName(string $identifier): string
+    public function getAttributeFriendlyName(string $identifier, string $type): string
     {
         $attributes = $this->getAttributeTypeAttributes();
         foreach ($attributes ?? [] as $attribute) {
             if ($identifier === $attribute->getName()) {
-                return $attribute->getLabel();
+                return $this->getLabelFromAttribute($attribute, $type);
             }
         }
         throw new AttributeNotFoundException(sprintf('Unable to find attribute with identifier: %s', $identifier));
     }
+
+    private function getLabelFromAttribute(
+        Attribute $attribute,
+        string $entityType
+    ) {
+        switch ($entityType) {
+            case Constants::TYPE_SAML:
+                return $attribute->getSaml20Label();
+            case Constants::TYPE_OPENID_CONNECT_TNG:
+                return $attribute->getOidcngLabel();
+            default:
+                return '';
```

Comment:
1. Returning an empty string when the entity type is not supported feels like a situation to create strange situations. This look like a great situation to be throw an exception instead. (also goes for the other getters below)
2. Please add a return type definition on these private methods.